### PR TITLE
test(STONEO11Y-52): run promtool linter in ci

### PR DESCRIPTION
**Pull request by yftacherzog**
Change the test script to support also lint tests, and have the CI
trigger it once for testing the rules and once for test linting.

Depends on #9

### CODE REVIEW

File: automation/test_prometheus.sh [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    prepare)
+        install_promtool
+        install_yq
+
+        echo "Extract Prometheus rules"
+        ./yq ".spec" $rule_file > $extracted_rule_file
+        ;;
+
+    lint)
+        echo "Running Prometheus linter tests"
```
Comment (srivickynesh) [inline automation/test_prometheus.sh:20]:
```suggestion
        echo "Running Prometheus rules linter tests"
```

File: .tekton/pull-request.yaml [hunk 2]
```diff
@@ -49,7 +49,36 @@ spec:
...
+          - prepare-for-prometheus-tests
+        taskSpec:
+          workspaces:
+            - name: workspace
+          steps:
+            - name: run-tests
+              image: registry.access.redhat.com/ubi8/ubi:latest
+              workingDir: $(workspaces.workspace.path)
+              script: ./automation/test_prometheus.sh test_rules
+      - name: run-prometheus-lint-tests
```
Comment (srivickynesh) [inline .tekton/pull-request.yaml:68]:
```suggestion
      - name: run-prometheus-rules-lint-tests
```

File: .tekton/pull-request.yaml [hunk 3]
```diff
@@ -49,7 +49,36 @@ spec:
           steps:
             - name: run-tests
               image: registry.access.redhat.com/ubi8/ubi:latest
-              script: cd $(workspaces.workspace.path) && ./automation/test.sh
+              workingDir: $(workspaces.workspace.path)
+              script: ./automation/test_prometheus.sh prepare
+      - name: run-prometheus-unit-tests
```
Comment (srivickynesh) [inline .tekton/pull-request.yaml:54]:
```suggestion
      - name: run-prometheus-rules-unit-tests
```

File: .tekton/pull-request.yaml [hunk 4]
```diff
@@ -49,7 +49,36 @@ spec:
           steps:
             - name: run-tests
               image: registry.access.redhat.com/ubi8/ubi:latest
-              script: cd $(workspaces.workspace.path) && ./automation/test.sh
+              workingDir: $(workspaces.workspace.path)
+              script: ./automation/test_prometheus.sh prepare
```
Comment (eisraeli) [inline .tekton/pull-request.yaml:53]: 1. Do you mind changing the name of the file to something more clear like just `promtool.sh`  ?

2. Eventually, I hope we can transform these shell scripts to something with an actual testing framework....

Comment (amisstea) [reply .tekton/pull-request.yaml:53]: To your second point, I think a Makefile would be nice improvement over shell scripts like this. Alternatively, building a container image with our testing tools preinstalled would eliminate most of it. We shouldn't use this PR for such a change but it's something to consider.

**Review by amisstea — COMMENTED** [review]
Looks good as long as 2314460792fecb53681a89259de535fe5c5ce0da is removed first. It seems you opened #9 for that.

File: automation/test_prometheus.sh (hunk 1 above)
Comment (yftacherzog) [reply automation/test_prometheus.sh]: done

File: .tekton/pull-request.yaml (hunk 3 above)
Comment (yftacherzog) [reply .tekton/pull-request.yaml]: done

**Comment by yftacherzog** [comment]
> Looks good as long as [2314460](https://github.com/redhat-appstudio/o11y/commit/2314460792fecb53681a89259de535fe5c5ce0da) is removed first. It seems you opened #9 for that.

Yes. I'll rebase on main once #9 is merged.

**Review by eisraeli — APPROVED** [review]
LGTM

File: .tekton/pull-request.yaml (hunk 4 above)
Comment (yftacherzog) [reply .tekton/pull-request.yaml]: alright

**Review by Omeramsc — APPROVED** [review]
LGTM

**Pull request merged**